docs/product-assembly_pam-on-openshift/src/main/asciidoc/environment-authoring-ha-proc.adoc

```diff
@@ -43,6 +43,7 @@ If you want to place the built KJAR files into an external Maven repository, set
 ** *Maven repository username* (`MAVEN_REPO_USERNAME`): The username for the Maven repository. 
 ** *Maven repository password* (`MAVEN_REPO_PASSWORD`): The username for the Maven repository. 
 +
+include::externaldb-params.adoc[]
```

Comment:
Yeah, I know what you mean. This goes against the approach for modules being standalone and being "include"-ed only in assemblies, but it's a valid need to include snippets of data that is used repeatedly. I've faced the same. I remember in Madcap Flare, there are specifically "snippet" files for this very reason. The way I've found to get around this is either to restructure so that whatever is being snippeted like this is converted/reworked as a separate section that is referenced, or I use ifevals to make the same content appear each time and only change what's different according to the section/doc it appears in (kind of the opposite of the snippeted include approach, which includes what's the same). The simple include route is certainly cleaner in the adoc file than the ifeval route, but breaks the modular structure and strategy. The ifeval route maintains the modular doc standard while ultimately accomplishing the same thing, but does require a bit more initial footwork to set up the file's content and is less clean.

I'm happy to chat more with you sometime about ifevals, and I'll think on this specific case some more myself too, since it's a valid issue. Meanwhile, here's some more info on ifevals: https://asciidoctor.org/docs/user-manual/#ifeval-directive. You can see examples in my migration assembly too if you'd like.

Btw, I'm not saying you need to make the change at this point (obviously there's not time now). Just that it should't stay like this indefinitely and I've given some alternatives to consider.